esx_service/utils/auth_api.py

```diff
@@ -449,19 +515,17 @@ def _tenant_update(name, new_name=None, description=None, default_datastore=None
             error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
             return error_info
     if default_datastore:
-        error_info = check_datastore(default_datastore)
+        error_info = check_default_datastore(default_datastore)
         if error_info:
             return error_info
-
-        datastore_url = vmdk_utils.get_datastore_url(default_datastore)
-        error_msg = tenant.set_default_datastore(auth_mgr.conn, datastore_url)
-        if error_msg:
-            error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
+        error_info = set_default_ds_and_create_privilege(tenant=tenant,
+                                                         default_datastore=default_datastore,
+                                                         check_existing=True)
+        if error_info:
```

Comment:
This function sometimes return error_info and None, while sometimes in just returns error_info.
The return should be uniform depending on the consumption of this function.